python/ray/data/datasource/image_folder_datasource.py

```diff
@@ -76,24 +79,34 @@ class ImageFolderDatasource(BinaryDatasource):
     def create_reader(
         self,
         root: str,
-        size: Tuple[int, int],
+        size: Optional[Tuple[int, int]] = None,
+        mode: Optional[str] = None,
     ) -> "Reader[T]":
-        """Return a :py:class:`Reader` that reads images.
+        """Return a :py:class:`~ray.data.datasource.Reader` that reads images.
+
+        .. warning::
+            If your dataset contains images of varying sizes and you don't specify
+            ``size``, this datasource will error. To prevent errors, specify ``size``
+            or :ref:`disable tensor extension casting <disable_tensor_extension_casting>`.
 
         Args:
             root: Path to the dataset root.
-            size: The desired height and width of loaded images.
+            size: The desired height and width of loaded images. If unspecified, images
+                retain their original shape.
+            mode: A `Pillow mode <https://pillow.readthedocs.io/en/stable/handbook/concepts.html#modes>`_
+                describing the desired type and depth of pixels. If unspecified, image
+                modes are inferred by
+                `Pillow <https://pillow.readthedocs.io/en/stable/index.html>`_.
 
         Raises:
             ValueError: if ``size`` contains non-positive numbers.
```

Comment:
shall we add comment for what's the error when user-provided `mode` is not correct?

Reply:
Fixed